airbyte-cdk/python/airbyte_cdk/sources/file_based/file_types/unstructured_parser.py

```diff
@@ -83,11 +90,25 @@ def parse_records(
     ) -> Iterable[Dict[str, Any]]:
         format = _extract_format(config)
         with stream_reader.open_file(file, self.file_read_mode, None, logger) as file_handle:
-            markdown = self._read_file(file_handle, file, format, logger)
-            if markdown is not None:
+            try:
+                markdown = self._read_file(file_handle, file, format, logger)
+                if markdown is not None:
+                    yield {
+                        "content": markdown,
+                        "document_key": file.uri,
+                        "error": None,
+                    }
+            except RecordParseError as e:
+                # pass explicitly raised RecordParseError through to be handled by the caller
+                raise e
+            except Exception as e:
+                exception_str = str(e)
+                logger.warn(f"File {file.uri} caused an error during parsing: {exception_str}.")
+                # catch unexpected exceptions during parsing and return them as part of the document
                 yield {
-                    "content": markdown,
+                    "content": None,
                     "document_key": file.uri,
+                    "error": exception_str,
                 }
```

Comment:
I worry a bit about proceeding on failure without the user opting into this behavior. 

Did you consider an option of another setting like `Ignore files with parsing errors`? I don't love adding more settings for the user to go through, but the ignore-errors-by-default behavior seems a bit odd to me here.

Reply:
Should we just wrap this up with the "skip unprocessable file types" (and maybe renaming to "skip unprocessable files"?)

In my mind it's a similar issue - it means that it's OK if not all files can be extracted, it's more about getting as much as possible.